llvm/lib/Target/AMDGPU/GCNSubtarget.cpp

```diff
@@ -748,6 +748,10 @@ GCNUserSGPRUsageInfo::GCNUserSGPRUsageInfo(const Function &F,
     FlatScratchInit = true;
   }
 
+  if (!AMDGPU::isGraphics(CC) && !IsKernel &&
+      !F.hasFnAttribute("amdgpu-no-lds-kernel-id"))
+    LDSKernelId = true;
+
```

Comment:
The kernel ID has to exist for all callable functions, !graphics will be wrong for amdgpu_gfx. Should be not entry function cc?

Reply:
I've just removed the refactoring for LDSKernelID since it's not relevant for kernels, but FYI this just copied the existing logic from `SIMachineFunction.cpp`, so if it's incorrect we should update it there.